geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -607,6 +650,84 @@ private void addIfMatching(GlobPattern matchPattern, List<byte[]> resultList, by
     }
   }
 
+  private RedisSortedSet getIntersection(List<RedisSortedSet> sets, ZAggregator aggregator) {
+    RedisSortedSet retVal = new RedisSortedSet(Collections.emptyList(), new double[] {});
+    RedisSortedSet smallestSet = sets.get(0);
+
+    for (RedisSortedSet set : sets) {
+      if (set.getSortedSetSize() < smallestSet.getSortedSetSize()) {
+        smallestSet = set;
+      }
+    }
+
+    for (byte[] member : smallestSet.members.keySet()) {
+      Double newScore;
+      if (aggregator.equals(ZAggregator.SUM)) {
+        newScore = getSumOfScoresForMember(sets, member, retVal);
+      } else if (aggregator.equals(ZAggregator.MAX)) {
+        newScore = getMaxScoreForMember(sets, member, retVal);
+      } else {
+        newScore = getMinScoreForMember(sets, member, retVal);
+      }
+
+      if (newScore != null) {
+        if (newScore.isNaN()) {
+          throw new ArithmeticException(ERROR_OPERATION_PRODUCED_NAN);
+        }
+        retVal.memberAdd(member, newScore);
+      }
+    }
+    return retVal;
+  }
+
+  private Double getSumOfScoresForMember(List<RedisSortedSet> sets, byte[] member,
```

Comment:
Oh you needed to use Double so you could return null. I think with the idea that we are computing the intersection on "this" (i.e. the retVal parameter goes away) and since these three methods will become one (by passing the aggregator into it) instead of it returning a Double/double it can return void and when it would have returned a non-null total it can instead check it for NAN and then just store it in this (i.e. memberAdd(member, total)).